packages/cms-lib/lib/walk.js

```diff
@@ -1,38 +1,74 @@
 const fs = require('fs');
 const path = require('path');
+const { logger } = require('../logger');
+const { logFileSystemErrorInstance } = require('../errorHandlers');
 
-function walk(dir) {
-  const processFiles = files => {
-    return Promise.all(
-      files.map(file => {
-        return new Promise((resolve, reject) => {
-          const filepath = path.join(dir, file);
-          fs.stat(filepath, (error, stats) => {
-            if (error) {
-              reject(error);
-            }
-            if (stats.isDirectory()) {
-              walk(filepath).then(resolve);
-            } else if (stats.isFile()) {
-              resolve(filepath);
-            }
+const STAT_TYPES = {
+  FILE: 'file',
+  SYMBOLIC_LINK: 'symlink',
+  DIRECTORY: 'dir',
+};
+
+const generateFilePromise = (dir, file) => {
+  return new Promise((resolve, reject) => {
+    const filepath = path.join(dir, file);
+    fs.lstat(filepath, (error, stats) => {
+      if (error) {
+        reject(error);
+      }
+      if (stats.isSymbolicLink()) {
+        resolve({
+          filepath,
+          type: STAT_TYPES.SYMBOLIC_LINK,
+        });
+      }
+      if (stats.isDirectory()) {
+        walk(filepath).then(fileContents => {
+          return resolve({
+            filepath,
+            type: STAT_TYPES.DIRECTORY,
+            fileContents,
```

Comment:
I don't think that `fileContents` is the best name here. I think either `files` or `folderContent` is more accurate.